### PR TITLE
#8131 Fix BigInteger and BigDecimal serialization by removing BigDecimalBivalue and BigIntegerBivalue

### LINKED ISSUE #7712: BiValue is redundant
With #7663 merged in it seems that all instances of `BiValue` have become in principle redundant.

All other `BiValue` wrappers were wrapping immutable  (in the sense of the specific underlying Java object they point to) instances of `IRubyObject`.

Classes like `org.jruby.RubyFloat` do not need this wrapper, they are themselves a wrapper for the Java object/value already.

In the example of `org.jruby.RubyFloat` there is already a field:

```java
    private final double value;
```

and an accessor:

```java
    /** Getter for property value.
     * @return Value of property value.
     */
    public double getValue() {
        return this.value;
    }
```

available. All our wrapper does here is potentially messing with the JIT's ability to inline things in our hot path (mind that our stacks are already very deep from JRuby itself) and create at least one extra object per `Event` field. 
With `StringBiValue` not caching the Java `String` anymore for reasons laid out in #7662 it seems reasonable to simply drop the `BiValue` path.

This will require however implementing the `SerializationProxy` logic that comes with `BiValueCommon` in some other way that can directly work with the Ruby types (it should be fairly trivial to make the CBOR and JSON serializers work with the Ruby objects, given that all the Ruby types we're dealing with here are in fact properly `Serializable` already).

Thoughts? :)

### LINKED ISSUE #8131: BigInteger are not Correctly Serialized
Same as #8128 pretty much.

Trying to send a `BigInteger` through a serialize, then deserialize cycle runs into:

```java
        Event e = new Event();
        final BigInteger huge =
            new BigInteger(String.valueOf(Long.MAX_VALUE) + String.valueOf(Long.MAX_VALUE));
        e.setField("bigint", huge);
        Event er = Event.deserialize(e.serialize());
        assertEquals(huge, er.getField("bigint"));
```

```
java.lang.IllegalArgumentException: Missing Converter handling for full class name=[B, simple name=byte[]

	at org.logstash.bivalues.BiValues.newBiValue(BiValues.java:32)
	at org.logstash.Valuefier.fallbackConvert(Valuefier.java:91)
	at org.logstash.Valuefier.convert(Valuefier.java:67)
	at org.logstash.ConvertedMap.newFromMap(ConvertedMap.java:50)
	at org.logstash.Event.<init>(Event.java:60)
	at org.logstash.Event.fromSerializableMap(Event.java:193)
	at org.logstash.Event.deserialize(Event.java:392)
	at org.logstash.EventTest.toBinaryRoundtrip(EventTest.java:53)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

The native-Java-serialization approach using `BiValue` or similar simply doesn't work with Jackson like this. For an `Object` of arbitrary bytes, Jackson simply goes ahead and serializes the fields in the `Object`, in this case `byte[]`.

`BigInteger` needs proper custom serialization here too, like implemented for `Timestamp` in #8130.

@jordansissel FYI, this isn't just a `Timestamp` issue, I think with the exception of `RubySymbol`, `Long` and `Integer` (and their counterparts) this breaks all remaining serializations via `BiValue`.

### PR DESCRIPTION
**Pull request by original-brownbear**
fixes #8131 and also effectively fixes #7712 (only the nil case and Java wrapper case ( #8132 ) ) left.

### CODE REVIEW

File: logstash-core/src/main/java/org/logstash/ObjectMappers.java [hunk 1]
```diff
@@ -30,14 +33,22 @@
             .addSerializer(RubySymbol.class, new RubySymbolSerializer())
             .addSerializer(RubyFloat.class, new RubyFloatSerializer())
             .addSerializer(RubyBoolean.class, new RubyBooleanSerializer())
-            .addSerializer(RubyFixnum.class, new RubyFixnumSerializer());
+            .addSerializer(RubyFixnum.class, new RubyFixnumSerializer())
+            .addSerializer(RubyBigDecimal.class, new RubyBigDecimalSerializer())
+            .addSerializer(RubyBignum.class, new RubyBignumSerializer());
+
+    private static final SimpleModule CBOR_DESERIALIZERS =
```
Comment (jakelandis) [inline logstash-core/src/main/java/org/logstash/ObjectMappers.java]: NIT: This should be `RUBY_DESERIALIZERS`, not `CBOR_SERIALIZERS`

Comment (original-brownbear) [reply logstash-core/src/main/java/org/logstash/ObjectMappers.java]: @jakelandis I wanted to make sure that it's clear this is only used with CBOR, never with JSON. I guess I could also JavaDoc it, but it seemed more straightforward for people to read this way right ?

Comment (jakelandis) [reply logstash-core/src/main/java/org/logstash/ObjectMappers.java]: Oh, I see now , the mapper it is registered to also carries the CBOR name.  in that case the name makes sense as-is.

**Comment by original-brownbear** [comment]
@jakelandis thanks merging :)

**Comment by colinsurprenant** [comment]
Per #8379 and #8131 the `BigInteger` and `BigDecimal` serialization problems were in fact related to the lack of CBOR **deserialization** support for these types in Jackson prior to version 2.8 and we  are using 2.7.  We are in the process of upgrading Jackson #7615 and once completed we can probably get rid of the custom serializers introduced here.

**Comment by jakelandis** [comment]
@colinsurprenant - I logged https://github.com/elastic/logstash/issues/8387 to address this.

**Comment by colinsurprenant** [comment]
and #8405 fixes the custom serializers

**Pull request closed without merging**